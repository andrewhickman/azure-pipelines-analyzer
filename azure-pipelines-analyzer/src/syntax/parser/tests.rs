use insta::assert_debug_snapshot;

use super::Parser;

macro_rules! case {
    ($method:ident($source:expr)) => {{
        let mut parser = Parser::new($source);
        parser.$method();
        assert_debug_snapshot!(parser.finish());
    }};
}

#[test]
pub fn parse_directive() {
    case!(directive("%YAML 1.2"));
    case!(directive("%YAML 1.2#"));
    case!(directive("%YAML 1.2 #comment"));
    case!(directive("%YAML 1.2 #comment\n\n#comment2\n"));
    case!(directive("foo"));
    case!(directive("%FOO"));
    case!(directive("%FOO \n"));
}
