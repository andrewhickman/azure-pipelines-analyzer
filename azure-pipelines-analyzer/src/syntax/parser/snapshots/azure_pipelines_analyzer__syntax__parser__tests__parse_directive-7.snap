---
source: azure-pipelines-analyzer/src/syntax/parser/tests.rs
expression: parser.finish()
---
Parse {
    node: Directive@0..4
      DirectiveToken@0..1 "%"
      DirectiveName@1..4 "FOO"
    ,
    errors: [],
}
